Sources/XiEditor/XiDocument.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+
+protocol XiDocument: class {
```

Comment:
I wonder if this might more accurately be thought of as a `XiView`? resize, for instance, isn't really a document concern. In any case this is a subtle distinction, in our current architecture.

Reply:
That's an interesting point. Do you think all of the rpc calls connected to document could be also a part of `XiDocument`?

Reply:
basically: I think that any RPC that takes as a parameter a `ViewIdentifier` should be part of this; those are the commands which are addressed to a specific document/view.